### PR TITLE
Frontend generator generates types/documents/hooks for use in useQuery

### PR DESCRIPTION
**Pull request by etha7**

### CODE REVIEW

File: frontend/package.json [hunk 1]
```diff
@@ -4,6 +4,9 @@
   "private": true,
   "dependencies": {
     "@apollo/client": "^3.5.10",
+    "@graphql-codegen/typed-document-node": "^2.2.8",
+    "@graphql-codegen/typescript": "^2.4.8",
+    "@graphql-codegen/typescript-operations": "^2.3.5",
```
Comment (normanjli) [inline frontend/package.json:9]: can we save these as dev dependencies? unless we are going to use a build pipeline that will require these. 
yarn remove etc...
yarn add etc... -d

**Review by normanjli — CHANGES_REQUESTED** [review]
Can we the add the codegen cli tools to the devDependencies? if not no worries

File: frontend/package.json (hunk 1 above)
Comment (etha7) [reply frontend/package.json:9]: its possible we won't be committing the generated code (will regenerate on each deployment), in which case we'll need the generator in production

Comment (normanjli) [reply frontend/package.json:9]: Fair point, would this be the same in the back end as well? Currently the backend has the code gen cli in the dev dependencies

Comment (etha7) [reply frontend/package.json:9]: good idea, but want to do in different pr because I'm lazy as hell

Comment (normanjli) [reply frontend/package.json:9]: You say lazy, its just good separation. No scope creep on this PR

**Pull request merged**